src/libraries/Microsoft.Extensions.HostFactoryResolver/src/HostFactoryResolver.cs

```diff
@@ -163,9 +163,8 @@ static bool IsApplicationNameArg(string arg)
                         => arg.Equals("--applicationName", StringComparison.OrdinalIgnoreCase) ||
                             arg.Equals("/applicationName", StringComparison.OrdinalIgnoreCase);
 
-                    args = args.Any(arg => IsApplicationNameArg(arg)) || assembly?.GetName().Name is null
-                        ? args
-                        : args.Concat(new[] { "--applicationName", assembly.GetName().Name }).ToArray();
+                    if (!args.Any(arg => IsApplicationNameArg(arg)) && assembly?.GetName().Name is string assemblyName)
+                        args = args.Concat(new[] { "--applicationName", assemblyName }).ToArray();
```

Comment:
This looks changing the logic, the original one can have `!` operator 
```suggestion
                    args = args.Any(arg => IsApplicationNameArg(arg)) || assembly.GetName().Name is null
                        ? args
                        : args.Concat(new[] { "--applicationName", assembly.GetName().Name! }).ToArray();
```

Reply:
Ah, sorry after looking it closely I see I was wrong. Lets remove the unnecessary `?` and add braces  
```suggestion
                    if (!args.Any(arg => IsApplicationNameArg(arg)) && assembly.GetName().Name is string assemblyName)
                    {
                        args = args.Concat(new[] { "--applicationName", assemblyName }).ToArray();
                    }
```